### PR TITLE
RK-7609 - Controller - support TLS mode

### PR DESCRIPTION
**Pull request by adilrookout**

### CODE REVIEW

File: charts/controller/README.md [hunk 1]
```diff
@@ -57,10 +57,23 @@ Those commands removes all the Kubernetes components associated with the chart a
 
 ## Configuration
 
+### Server Modes
+
+The controller runs with one of 3 modes (controller.serverMode):
```
Comment (Gilwe) [inline charts/controller/README.md:62]: 2 modes lol

File: charts/controller/README.md [hunk 2]
```diff
@@ -57,10 +57,23 @@ Those commands removes all the Kubernetes components associated with the chart a
...
+
+* **TLS** - If you have your own certificate that resides also for your teammates browser, you will need to create :
```
Comment (Gilwe) [inline charts/controller/README.md:64]: The reason for TLS here isn't like our DOP.
The cert doesn't need to reside in the users browser, as they use it only for the sdk setup.

File: charts/controller/templates/deployment.yaml [hunk 3]
```diff
@@ -65,3 +67,23 @@ spec:
...
+          - mountPath: /var/controller-tls-secrets/tls.crt
+            subPath: tls.crt
+            name: tls-cert
+            readOnly: true
+{{ end }}
+{{ if (eq .Values.controller.serverMode "TLS") }}
+      volumes:
+      - name: tls-key
+        secret:
+          secretName: rookout-tls-key
```
Comment (Gilwe) [inline charts/controller/templates/deployment.yaml:85]: use controller.tlsKeySecretName and controller.tlsCertificateConfigmapName

**Comment by Gilwe** [comment]
Also, update the `version` in both the controller's and datastore's `chart.yaml`

File: charts/controller/templates/deployment.yaml (hunk 3 above)
Comment (adilrookout) [reply charts/controller/templates/deployment.yaml:85]: I wanted to hardcode it so the customer won't need to pass it to the template, just define them in k8s

Comment (Gilwe) [reply charts/controller/templates/deployment.yaml:85]: so what's the usage of the variables `controller.tlsKeySecretName` and `controller.tlsCertificateConfigmapName` that you added?

Comment (adilrookout) [reply charts/controller/templates/deployment.yaml:85]: removed ... :)

**Review by Gilwe — DISMISSED** [review]

**Comment by ElDuderinos** [comment]
@adilrookout can we merge this?

**Review by Gilwe — APPROVED** [review]

**Pull request merged**